generator/src/main/java/com/scottlogic/deg/generator/Guice/ProfileProvider.java

```diff
@@ -0,0 +1,33 @@
+package com.scottlogic.deg.generator.Guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.scottlogic.deg.generator.CommandLine.GenerateCommandLine;
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.inputs.InvalidProfileException;
+import com.scottlogic.deg.generator.inputs.ProfileReader;
+import com.scottlogic.deg.generator.inputs.validation.ProfileValidator;
+
+import java.io.IOException;
+
+public class ProfileProvider implements Provider<Profile> {
+    private final GenerateCommandLine commandLine;
+    private final ProfileReader profileReader;
+    private Profile profile;
```

Comment:
This shouldn't be a field, it should (at most) be local to the `get` method. Providers shouldn't maintain any state.